packages/fuels-core/src/types/tx_status.rs

```diff
@@ -65,50 +72,61 @@ impl TxStatus {
             Self::SqueezedOut(SqueezedOut { reason }) => {
                 Err(Error::Transaction(Reason::SqueezedOut(reason.clone())))
             }
-            Self::Revert(Revert {
+            Self::Failure(Failure {
+                receipts,
+                reason,
+                revert_id,
+                ..
+            })
+            | Self::PreconfirmationFailure(Failure {
                 receipts,
                 reason,
-                revert_id: id,
+                revert_id,
                 ..
-            }) => Err(Self::map_revert_error(&receipts, &reason, id, log_decoder)),
+            }) => Err(Self::map_revert_error(
+                receipts,
+                &reason,
+                revert_id,
+                log_decoder,
+            )),
             Self::Submitted => Err(Error::Transaction(Reason::Other(
                 "transactions was not yet included".to_owned(),
             ))),
-            Self::Success(success) => Ok(success),
+            Self::Success(success) | Self::PreconfirmationSuccess(success) => Ok(success),
         }
     }
 
     pub fn total_gas(&self) -> u64 {
         match self {
             TxStatus::Success(Success { total_gas, .. })
-            | TxStatus::Revert(Revert { total_gas, .. }) => *total_gas,
+            | TxStatus::Failure(Failure { total_gas, .. }) => *total_gas,
             _ => 0,
         }
     }
 
     pub fn total_fee(&self) -> u64 {
         match self {
             TxStatus::Success(Success { total_fee, .. })
-            | TxStatus::Revert(Revert { total_fee, .. }) => *total_fee,
+            | TxStatus::Failure(Failure { total_fee, .. }) => *total_fee,
             _ => 0,
         }
     }
 
     fn map_revert_error(
-        receipts: &[Receipt],
+        receipts: Vec<Receipt>,
```

Comment:
Why do we take ownership of the receipts here? We only use references, and I see that this is causing redundant clone operations in the code.

Reply:
At the end of the function we create a `Error::Transaction(Reason::Failure` that clones the receipts. My idea was if we need ownership of the receipts better request it in the function signature.